library/git/pull.sh

```diff
@@ -0,0 +1,49 @@
+#!/bin/bash
+
+set -x;
+
+ssh_user=${SSH_USER:-abhijith}
+builder=${BUILD_SYSTEM:-vapundevrhel7bld.vxindia.veritas.com}
+workspace=${WORKSPACE:-/lhome/${ssh_user}/workspace/sfnas/}
+
+ssh_builder="ssh ${ssh_user}@${builder}"
+
+# Depending on whether Appliance/Plugin Manager is running as a root or normal user, it'll switch to ssh_user, so that ssh commands can be run.
+run_as_ssh_user()
+{
+  cmd=$1
+  cur_user=$(id -un)
+  if [ "root" == "${cur_user}" ]; then
+    su -c "${ssh_builder} \"$cmd\"" -s /bin/bash ${ssh_user}
+  else
+    ${ssh_builder} "$cmd"
+  fi
+}
+
+# Logging user under which service is being run just for debugging purpose!
+echo "Local user..."
+su -c "id; pwd" -s /bin/bash ${ssh_user}
+
+cur_branch=$(run_as_ssh_user "cd ${workspace}; git branch --list | grep -F \\\"*\\\"")
```

Comment:
There are more direct ways of discovering the name of the current branch (assuming there _is_ a current branch).

```suggestion
cur_branch=$(run_as_ssh_user "cd ${workspace}; git rev-parse --abbrev-ref HEAD)
```